### PR TITLE
Add remote-info command

### PR DESCRIPTION
**Pull request by alexlarsson**
This shows information about a ref in a remote. Of particular interest is the --log option which gives you a history which can be used with update --commit=XXX to roll back an update.

### CODE REVIEW

File: app/flatpak-builtins-info-remote.c [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+
+              p_commit_v = flatpak_dir_fetch_remote_commit (dir, remote, ref, p, NULL, cancellable, NULL);
+              if (p_commit_v == NULL)
+                break;
+
+              p_parent = ostree_commit_get_parent (p_commit_v);
+              p_timestamp = ostree_commit_get_timestamp (p_commit_v);
+              p_formated_timestamp = format_timestamp (p_timestamp);
+
+              g_variant_get (commit_v, "(a{sv}aya(say)&s&stayay)", NULL, NULL, NULL,
```
Comment (mwleeds) [inline app/flatpak-builtins-info-remote.c]: should be `p_commit_v`

File: doc/flatpak-remote-info.xml [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+    </refentryinfo>
+
+    <refmeta>
+        <refentrytitle>flatpak remote-info</refentrytitle>
+        <manvolnum>1</manvolnum>
+    </refmeta>
+
+    <refnamediv>
+        <refname>flatpak-remote-info</refname>
+        <refpurpose>Show information about installed application and/or runtime in a remote</refpurpose>
```
Comment (mwleeds) [inline doc/flatpak-remote-info.xml]: Remove "installed"

File: doc/flatpak-remote-info.xml [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+                <arg choice="plain">REMOTE</arg>
+                <arg choice="plain">REF</arg>
+            </cmdsynopsis>
+    </refsynopsisdiv>
+
+    <refsect1>
+        <title>Description</title>
+
+        <para>
+            Shows information about a runtimes and applications from the
```
Comment (mwleeds) [inline doc/flatpak-remote-info.xml]: Replace with `the runtime or application <arg choice="plain">REF</arg>`

File: doc/flatpak-remote-info.xml [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+            remote repository with the name <arg choice="plain">REMOTE</arg>.
+            You can find all configured remote repositories with flatpak remote-list.
```
Comment (mwleeds) [inline doc/flatpak-remote-info.xml]: remote-list → remotes

File: doc/flatpak-remote-info.xml [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+                    Using <arg choice="plain">--installation=default</arg> is equivalent to using
+                    <arg choice="plain">--system</arg>.
+                </para></listitem>
+            </varlistentry>
+
+            <varlistentry>
+                <term><option>--runtime</option></term>
+
+                <listitem><para>
+                    Assume that all <arg choice="plain">REF</arg>s are runtimes if not explicitly specified.
```
Comment (mwleeds) [inline doc/flatpak-remote-info.xml]: there's only one REF

File: doc/flatpak-remote-info.xml [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+                </para></listitem>
+            </varlistentry>
+
+            <varlistentry>
+                <term><option>--app</option></term>
+
+                <listitem><para>
+                    Assume that all <arg choice="plain">REF</arg>s are apps if not explicitly specified.
```
Comment (mwleeds) [inline doc/flatpak-remote-info.xml]: there's only one REF

File: doc/flatpak-remote-info.xml [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+</programlisting>
+
+    </refsect1>
+
+    <refsect1>
+        <title>See also</title>
+
+            <para>
+                <citerefentry><refentrytitle>flatpak</refentrytitle><manvolnum>1</manvolnum></citerefentry>,
+                <citerefentry><refentrytitle>flatpak-remotes</refentrytitle><manvolnum>1</manvolnum></citerefentry>
```
Comment (mwleeds) [inline doc/flatpak-remote-info.xml]: A link to the `remote-ls` man page would make sense

File: app/flatpak-builtins-info-remote.c [hunk 8]
```diff
@@ -0,0 +1,328 @@
...
+  g_autofree char *branch = NULL;
+  g_auto(GStrv) parts = NULL;
+  FlatpakKinds kind;
+  g_autofree char *ref = NULL;
+  g_autofree char *commit = NULL;
+  g_autofree char *parent = NULL;
+  const char *on = "";
+  const char *off = "";
+  gboolean friendly = TRUE;
+  const char *xa_metadata = NULL;
```
[the remark this answers is not in the document]
Comment (alexlarsson) [reply app/flatpak-builtins-info-remote.c]: Metadata is a bit large, so i think its better to use &s for the get arg.

File: app/flatpak-builtins-info-remote.c [hunk 9]
```diff
@@ -0,0 +1,328 @@
...
+  if (argc < 3)
+    return usage_error (context, _("REMOTE and REF must be specified"), error);
+
+  remote = argv[1];
+  pref = argv[2];
+
+  default_branch = flatpak_dir_get_remote_default_branch (dir, remote);
+  kinds = flatpak_kinds_from_bools (opt_app, opt_runtime);
+
+  if (!flatpak_split_partial_ref_arg (pref, kinds, opt_arch, target_branch,
```
[the remark this answers is not in the document]
Comment (alexlarsson) [reply app/flatpak-builtins-info-remote.c]: default, good catch.

[the remark this answers is not in the document]
Comment (alexlarsson) [reply app/flatpak-builtins-info-remote.c]: Actually, no, it should be NULL i think.

[the remark this answers is not in the document]
Comment (alexlarsson) [reply app/flatpak-builtins-info-remote.c]: The default is handled in the find_remote_ref. This is if you manually specified a particular branch, which you can only do in the partial-ref itself in this case.

**Comment by alexlarsson** [comment]
Had to rebase to fix the merge issues

**Comment by alexlarsson** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**